sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,631 @@
+
+        Args:
+
+        Returns:
+            One of the following predefined constants:
+               SWITCHING_MODE_AUTO if auto switch is enabled.
+               SWITCHING_MODE_MANUAL if manual switch is enabled.
+        """
+
+        raise NotImplementedError
+
+    def get_nic_temperature(self):
+        """
+        This API specifically returns nic temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_temperature(self):
+        """
+        This API specifically returns local ToR temperature of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            an Integer, the temperature of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def get_nic_voltage(self):
+        """
+        This API specifically returns nic voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_voltage(self):
+        """
+        This API specifically returns local ToR voltage of the physical port for which this API is called.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a float, the voltage of the local MCU
+        """
+
+        raise NotImplementedError
+
+
+#############################################################################################
+###                                  Debug Functionality                                  ###
+#############################################################################################
+
+
+    def set_debug_mode(self, enable):
+        """
+        This API specifically enables/disables a debug mode that the port is now
+        going to be run on. If enabled, this means that PRBS/Loopback etc. type diagnostic mode
+        is now going to be run on the port and hence normal traffic will be disabled
+        on it if enabled and vice-versa if disabled.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            enable:
+            a boolean, True if the debug mode needs to be enabled
+                     , False if the debug mode needs to be disabled
+
+
+        Returns:
+            a boolean, True if the enable is successful
+                     , False if the enable failed
+        """
+
+        raise NotImplementedError
+
+    def get_debug_mode(self):
```

Comment:
Is it possible to have this synchronized across both ToR's? 
If not we might have to make this API as optional/or else remove it.

Reply:
@jleveque

Reply:
This was intended to be a software mechanism which would allow debug functionality if enabled and stop/disable debug functionality if disabled. However, these actions were all considered to be performed from the same ToR. Synchronization across ToRs was not considered.

@xinyulin: Do you foresee problems with this approach? Does the other ToR side need to be aware that debugging is occurring on the other side?

Reply:
@jleveque: The only concern is the debug mode status is not synchronized. In application, the other ToR side don't need to be aware that debugging mode of the other side. As you mentioned, it should be fine if all the debug actions were being performed from the same ToR.

@vdahiya12: I will consider to find a simple way to synchronize debug mode across ToRs for next firmware update